### ISSUE TITLE
CloudFormation region selection

### ISSUE THREAD

**Issue by oren-zohar** [elastic/cloudbeat#605]
## Summary
Provide the user a select list where he could choose multiple regions to deploy. 
Should we also have the option to select all at once?

## Multi-Region
How can we support multi-region collection?

**Deploy per region**
- Each deployment will collect each own resources

**Deploy in one region**
- Deployed in one region and collected all data from all regions
- How will this affect billing and resource usage?

**Comment by eyalkraft** [comment]
check out [this article](https://aws.amazon.com/blogs/infrastructure-and-automation/multiple-account-multiple-region-aws-cloudformation/)

**Comment by eyalkraft** [comment]
Also additional context for why we start with deploy per region-
The reason is that moving/reading snapshots across regions might be expensive.

## References

### Data transfer


[Data transfer explanation](https://aws.amazon.com/blogs/architecture/overview-of-data-transfer-costs-for-common-architectures/#:~:text=Accessing%20services%20across%20AWS%20Regions)

>If your workload accesses services in different Regions (Figure 2), there is a charge for data transfer across Regions. The charge depends on the source and destination Region (as described on the [Amazon EC2 Data Transfer pricing page](https://aws.amazon.com/ec2/pricing/on-demand/#Data_Transfer)).

[Data transfer costs table](https://aws.amazon.com/ec2/pricing/on-demand/#Data_Transfer)

### EBS

[EBS `GetSnapshotBlock` Pricing](https://aws.amazon.com/ebs/pricing/#:~:text=GetSnapshotBlock%20API%20is,SnapshotAPIUnits%20*%20100%2C000%20SnapshotAPIUnits)

> GetSnapshotBlock API is charged per SnapshotAPIUnit. One GetSnapshotBlock API request uses a block size of 512 KiB and consumes one SnapshotAPIUnit. Note that 512 KiB is the only block size supported today, so each GetSnapshotBlock request consumes one SnapshotAPIUnit. For example, if you make 100,000 GetSnapshotBlock API calls using a block size of 512 KiB in a region that charges $0.003 per thousand SnapshotAPIUnits, you will be charged $0.30 ($0.003 per thousand SnapshotAPIUnits * 100,000 SnapshotAPIUnits).

> Note: If you use external or cross-region data transfers, additional [EC2 data transfer](https://aws.amazon.com/ec2/pricing/) charges will apply.

[EBS Snapshot Pricing](https://aws.amazon.com/ebs/pricing/#:~:text=Example%207%20%E2%80%93%20EBS%20Snapshots)
> Copying EBS snapshots is charged for the data transferred across regions. After the snapshot is copied, standard EBS snapshot charges apply for storage in the destination region.

[Copy an Amazon EBS snapshot](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-copy-snapshot.html)

[Question - Copying EBS Snapshot Across Regions](https://repost.aws/questions/QUwHTYd0C9SmS3v4myPmUhNg/copying-ebs-snapshot-across-regions)

**Comment by amirbenun** [comment]
Thanks @eyalkraft, absolutely agree with the EC2 per region model.
A few options are available if we go in that direction:
## 1. Manual deployment per region
Let the user manually deploy the CloudFormation stack on every region he wants to activate the vulnerability scanning.
### Pros:
- Easy to develop and more practical for 8.8.
### Open questions:
- The quick creation link for CloudFormation should contain the AWS region. This probably means the user will have to select a region on Kibana to get a parameterized link. Can we ask for the deployment region on the elastic-agent deployment? (question for product)

## 2. Cross-region StackSet deployment
Deploy a CloudFormation template that deploys a [StackSet](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-concepts.html) on multiple regions under that account.
### Pros:
- More modular approach - we can create a single IAM role and attach it to multiple EC2 instances of different regions.
- Better user experience - single click deploys on multiple regions.
### Open questions:
- Can a CloudFormation template contain a StackSet? (requires a POC based on the [link](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudformation-stackset.html))
- How does the user specify the regions? CloudFormation parameters support `CommaDelimitedList` which means that the user will manually fill in something like `eu-west-1,eu-west-2`
Is that a good user experience? (question for product)
- Do we have region-specific settings? For example, setting a different instance type on two different regions may not be possible. (question for product)


@nick-alayil  I would like to hear your thoughts about the product open questions.

**Comment by eyalkraft** [comment]
@amirbenun Thanks for this clear explanation!
I lean towards option 2.

**Comment by amirbenun** [comment]
Another issue that we should consider is that for creating a stack set, the user needs to have an organization's master permissions. On a POC I got the following result:
```
You must be the master or delegated admin account of an organization before operating a SERVICE_MANAGED stack set
```

In the end, it may limit accounts from activating vulnerability scanning.

**Comment by nick-alayil** [comment]
As discussed in the product sync, we'll stick to the deployment per region model for MVP/8.8 release. During onboarding, the user will be asked to select the AWS region (only one region per policy) they're interested in. As a result, we will ensure seamless onboarding, but also keep MVP requirements to a minimum. 

This also gives us the opportunity to do POCs, analyzing the cost and performance of scanning all assets of all regions from one region. That being said, I'd recommend we tackle the further simplification (one-click installation for all regions) of onboarding only after learning the MVP customer feedbacks.

**Comment by amirbenun** [comment]
Thanks @nick-alayil that sounds like the right choice for the 8.8 release.
